### PR TITLE
Projects data collection

### PR DESCRIPTION
**Pull request by vidhiJain**
## Motivation and Context

- Created  `data collection` in `projects/` to collect and process h5 scripts. 

## How Has This Been Tested

On the Hello-Robot, data is collected using teleop assuming hector slam and joy node are running:
``` 
python projects/data_collection/collect_h5.py --task-name test
```
Then saved h5 files can be loaded by:
```
python projects/data_collection/h5_dataloader.py 
```

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [x] I have made changes to existing documentation where needed.
- [x] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

**Comment by vidhiJain** [comment]
How to load the `image_keys`? 
Here is the current output of h5_dataloader
```
~/home-robot$ python projects/data_collection/h5_dataloader.py 
{'temporal': {'q': tensor([[-0.6177,  0.1266,  1.6331,  0.2213,  0.0068,  0.0867, -0.0123, -0.2669,
          3.2942,  0.1006, -1.0987]], dtype=torch.float64), 'dq': tensor([[0.0000, 0.0000, 0.0000, 0.0002, 0.0004, 0.0000, 0.0000, -0.0000, -0.0000,
         -0.0000, -0.0000]], dtype=torch.float64), 'ee_pose': tensor([[-0.4921,  0.1019,  0.3259,  0.6133, -0.5026, -0.3787,  0.4774]],
       dtype=torch.float64), 'base_pose': tensor([[-6.1766e-01,  1.2665e-01,  1.6331e+00,  1.6779e+09]],
       dtype=torch.float64), 'camera_pose': tensor([[[ 1.3649e-01, -9.8806e-01, -7.1461e-02, -6.2103e-01],
         [-8.7871e-01, -1.5406e-01,  4.5181e-01,  1.7086e-01],
         [-4.5742e-01,  1.1258e-03, -8.8925e-01,  1.2753e+00],
         [ 0.0000e+00,  0.0000e+00,  0.0000e+00,  1.0000e+00]]],
       dtype=torch.float64), 'user_keyframe': tensor([[1]])}, 'image': {}}
```

File: projects/data_collection/tutorial_h5_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import os
+import numpy as np
+import torch
+
+from home_robot.utils.data_tools.loader import DatasetBase
+
+class SimpleDataset(DatasetBase):
+    """Simple example data loader."""
+
+    def get_datum(self, trial, idx):
```
Comment (zephirefaith) [inline projects/data_collection/tutorial_h5_dataloader.py:10]: add type hints here

**Review by zephirefaith — APPROVED** [review]

**Comment by zephirefaith** [comment]
This is great. Thanks for the detailed documentation and addition of the nice tutorial. 1 minor comment about adding type hints to the tutorial `get_datum` signature, but otherwise this is great. Please make that modification and we can merge this.

**Comment by zephirefaith** [comment]
Closing this. I will make the type hint change on my end. Thanks!

**Pull request merged**